src/Backend/Compile.hs

```diff
@@ -20,34 +25,74 @@ data Arguments = Arguments
   , outputFile :: !FilePath
   }
 
+minLlvmVersion :: Version
+minLlvmVersion = makeVersion [4, 0, 0]
+
+maxLlvmVersion :: Version
+maxLlvmVersion = makeVersion [5, 0, 1]
```

Comment:
Do we care about anything other than the major version in these? Perhaps they could just be the ints 4 and 5? That would simplify the `supportedLlvmVersions` function a bit.

Reply:
Version 6 will be out relatively soon (in March) and it isn't clear to me when the Ubuntu repositories will be updated.

Reply:
Also, I just realised this -- if the LLVM flag is supplied, we can check the version using these two bounds instead of using it directly. So it is better to have the type `Version` instead of `Int`.

Reply:
Issuing a warning if `llvm-config --version` is not in range seems like a good idea, but are you sure we need to care about anything other than major version even in that case?

Reply:
Nope, we can work with Ints, that's fine.

Reply:
I'll share a gist to show what it looks like...

Reply:
Oops, I just realised there is a minor version in the function too. Yes, I agree it is not needed in both places.